diff --git a/build/generateProperties.js b/build/generateProperties.js
index 305d3685440b08eca2f9f7fc456eade9df90163e..4cb671d6195cd56829de01a68edc22ae4ac311dd 100644
--- a/build/generateProperties.js
+++ b/build/generateProperties.js
@@ -76,14 +76,14 @@ const generateProperties = (details, generateFor, nonCompositeTypeImports, compo
             const x = compositeTypeImports[p.type.fullName];
             typeImports.push(x);
             zodType = x.name;
-            if (p.isArray)
-                zodType = `${zodType}.array()`;
-            if (p.isNullable)
-                zodType = `${zodType}.nullable()`;
         }
         else {
             console.error(`kanel-zod: Unknown type for ${name}.${p.name}: ${p.type.fullName}`);
         }
+        if (p.isArray)
+            zodType = `${zodType}.array()`;
+        if (p.isNullable)
+            zodType = `${zodType}.nullable()`;
         let isOptional;
         if (optionalOverride === undefined) {
             switch (generateFor) {
